File: elementsOfProgrammingInterviews/JS/ch06_arrays/generatePrimes.js
const generatePrimes = (n) => {
  let primes = [];
  let isPrime = new Array(n+1);
  for (let i = 0; i < isPrime.length; i++) {
    isPrime[i] = true;
  }
  
  isPrime[0] = false;
  isPrime[1] = false;
  
  for (let p = 2; p <= n; p++) {
    if (!!isPrime[p]) {
      primes.push(p);
      for (let j = p; j <= n; j+=p) {
        isPrime[j] = false;
      }
    }
  }
  
  return primes;
}

File: elementsOfProgrammingInterviews/CPP/ch06_arrays/generatePrimes.cpp
#include <iostream>
#include <vector>
using namespace std;

vector<int> generatePrimes(int n) {
  vector<int> primes;
  vector<bool> isPrime(primes.size()+1, true);
  isPrime[0] = false;
  isPrime[1] = false;
  for (int i = 2; i <= n; i++) {
    if (isPrime[i]) {
      primes.push_back(i);
      for (int j = i; j <= n; j += i) {
	isPrime[j] = false;
      }
    }
  }

  return primes;
}

int main() {
  vector<int> v = generatePrimes(10);
  for (int i : v) {
    cout << i;
  }
  cout << endl;
}
